base/src/main/java/io/spine/environment/EnvironmentType.java

```diff
@@ -45,6 +53,28 @@ public abstract class EnvironmentType {
      */
     protected abstract boolean enabled();
 
+    /**
+     * Installs the callback to be called when this environment type is
+     * {@linkplain #enabled() detected.}
+     */
+    final void onDetected(@Nullable Consumer<T> callback) {
```

Comment:
Looks like this one should be `public`. Otherwise it is inaccessible for the framework users.

Reply:
This method is intended for the framework users:
```java
public <T extends EnvironmentType<T>>
    void whenDetected(Class<T> cls, @Nullable Consumer<T> callback)
```
An instance method would require having an instance, which is the opposite to the "protocol" we have with working with `Environment`: we work via `Class` references instead.